logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java

```diff
@@ -113,10 +112,8 @@ public void close() throws IOException {
             }
         }
         IThrowableProxy throwableProxy = event.getThrowableProxy();
-        if (throwableProxy instanceof ThrowableProxy) {
-            EcsJsonSerializer.serializeException(builder, ((ThrowableProxy) throwableProxy).getThrowable(), stackTraceAsArray);
```

Comment:
Removing this branch increases allocations quite a bit (by serializing the stack trace into a `String` as opposed to a `StringBuidler`) for what is, I think, the usual case when no custom `throwableConverter` is set.

Could you restore that branch by checking if a custom throwable converter has been set?

Reply:
updated the code to only invoke the `throwableConverter` if one is set by the user. Hopefully this resolves the degradation in memory management that I introduced in the previous commits